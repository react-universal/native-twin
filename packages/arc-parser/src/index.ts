export * from './parsers/Parser';
export * from './parsers/between.parser';
export * from './parsers/choice.parser';
export * from './parsers/composed.parsers';
export * from './parsers/coroutine.parser';
export * from './parsers/data.parser';
export * from './parsers/debug.parser';
export * from './parsers/either.parser';
export * from './parsers/lookahead';
export * from './parsers/many.parser';
export * from './parsers/maybe.parser';
export * from './parsers/number.parser';
export * from './parsers/peek.parser';
export * from './parsers/recursive.parser';
export * from './parsers/separated-by.parser';
export * from './parsers/sequence-of';
export * from './parsers/skip.parser';
export * from './parsers/string.parser';
export * from './parsers/util.parsers';
export * from './parsers/functional.parsers';
export type {
  InternalResultType,
  ParserError,
  ParserState,
  ParserSuccess,
  ResultType,
  StateTransformerFunction,
} from './types';
