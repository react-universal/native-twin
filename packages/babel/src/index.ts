import { PluginObj } from '@babel/core';
import { addNamed } from '@babel/helper-module-imports';
import { PLUGIN_IMPORT_META } from './constants/plugin.constants';
import { isReactImport, isReactRequire } from './effects/path.effects';
import { createMemberExpressionProgram } from './effects/programs';
import { createVisitorContext } from './effects/visitor-context';
import { BabelAPI } from './types';

export default function nativeTwinBabelPlugin(babel: BabelAPI): PluginObj {
  // const t = babel.types;
  const createContext = createVisitorContext(babel.types);

  return {
    name: '@native-twin/native-twin-metro-imports',
    visitor: {
      MemberExpression(path, state) {
        const context = createContext(path, state);

        if (!context.stateContext.isValidPath) return;

        const shouldReplace = createMemberExpressionProgram(path);

        if (shouldReplace) {
          path.replaceWith(addNamed(path, ...PLUGIN_IMPORT_META));
          return;
        }
      },
      Identifier(path, state) {
        const context = createContext(path, state);
        if (!context.stateContext.isValidPath) return;
        if (path.node.name === 'createElement' && path.parentPath.isCallExpression()) {
          const binding = path.scope.getBinding(path.node.name);
          if (!binding) return;
          if (isReactRequire(binding) || isReactImport(binding)) {
            path.replaceWith(addNamed(path, ...PLUGIN_IMPORT_META));
          }
        }
      },
    },
  };
}
