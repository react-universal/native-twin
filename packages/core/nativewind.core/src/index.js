export { styled, useColorScheme, NativeWindStyleSheet, useTailwind, fontScale, roundToNearestPixel, getPixelSizeForLayoutSize, hairlineWidth, pixelRatio, platformColor, platformSelect, } from 'nativewind';
export { colors } from './colors';
export { theme } from './theme';
//# sourceMappingURL=index.js.map