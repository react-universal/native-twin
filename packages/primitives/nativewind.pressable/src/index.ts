export function primitivesNativewindPressable(): string {
    return 'primitives-nativewind.pressable';
}
