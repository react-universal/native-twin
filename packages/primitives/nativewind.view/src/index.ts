export * from './lib/primitives-nativewind.view';
