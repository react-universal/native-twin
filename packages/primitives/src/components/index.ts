export * from './View';
export * from './Nav';
export * from './Text';
export * from './Image';
export * from './Pressable';
