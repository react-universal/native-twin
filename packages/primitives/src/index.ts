export { View } from './components/View';
export type { ViewProps } from './components/View';
export { Nav } from './components/Nav';
export { Code, H1, H2, H3, H4, H5, H6, P, Span, Strong } from './components/Text';
export { Image } from './components/Image';
export type { IImageProps } from './components/Image';
export { Pressable } from './components/Pressable';
export { TextInput } from './components/TextInput';
export type { TextInputProps } from './components/TextInput';
export { TBody, TD, TFoot, TH, THead, TR, Table } from './components/Table';
export { LI, UL } from './components/List';
export { FlatList } from './components/FlatList';

// export { Rect, Svg } from './components/svg';
