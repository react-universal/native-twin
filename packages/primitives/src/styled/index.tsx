import { ComponentProps, ComponentType, forwardRef } from 'react';
import { parseClassNames } from '@react-universal/core';

function styled<P extends unknown>(Component: ComponentType<P>) {
  const Styled = forwardRef<unknown, ComponentProps<typeof Component>>(
    //@ts-expect-error
    ({ className = '', ...restProps }, ref) => {
      const result = parseClassNames(className);
      console.log('STYLES: ', result);
      return <Component {...restProps} style={result.styles ?? {}} ref={ref} />;
    },
  );
  if (typeof Component !== 'string') {
    Styled.displayName = `StyledTW.${Component.displayName || Component.name || 'NoName'}`;
  }

  return Styled;
}

export { styled };
