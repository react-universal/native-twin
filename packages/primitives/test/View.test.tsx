import renderer from 'react-test-renderer';
import { describe, expect, it } from 'vitest';
import { Span, View } from '../src';

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();
  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);
  return result as renderer.ReactTestRendererJSON;
}

describe('@universal-labs/primitives', () => {
  it('View render', () => {
    const component = renderer.create(
      <View className='flex-1'>
        <Span>Test View</Span>
      </View>,
    );
    let tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
