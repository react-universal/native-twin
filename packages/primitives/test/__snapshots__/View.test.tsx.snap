// Vitest Snapshot v1

exports[`@universal-labs/primitives > View render 1`] = `
<div
  className="css-view-175oi2r undefined flex-1"
  dir={null}
>
  <span>
    Test View
  </span>
</div>
`;
